privacyidea/lib/eventhandler/base.py

```diff
@@ -356,10 +398,19 @@ def check_condition(self, options):
                 if not ','.join([tok.get_serial() for tok in token_objects]):
                     return False
 
-        if "tokenrealm" in conditions and tokenrealms:
+        if CONDITION.TOKENREALM in conditions and tokenrealms:
```

Comment:
Hmm, I'm not sure about the logic of ``and tokenrealms`` here: Assume we have an event handler with condition ``realm=defrealm``, and we have a token without any realms. In a request with that token, the handler is executed, because ``tokenrealms`` is empty. I would expect the handler not to be executed, because the token realm is not ``defrealm``. What do you think?

Reply:
There are several of these conditions: "and serial", "and user".
The originial reason for this was not that the token would have no tokenrealm but rather, that the tokenrealm information is not available in this request/ could not be determined, since the serialnumber of a token was not specified or could not be determined.

If you think we should change this, this is absolutely ok. But I would prefer opening a new issue for that, so that it is easier to track the actual logical change in the behaviour.

Reply:
Ah okay, thanks for the clarification! I opened #1460 to track this issue.